src/datautils/schematicShapeGenerator.ts

```diff
@@ -788,35 +801,18 @@ const createSubkindPerforationTexture = (
   return texture;
 };
 
-/**
- * Yellow gravel
- * @param perforationOptions
- * @returns
- */
-const createSubkindOpenHoleGravelPackTexture = (perforationOptions: PerforationOptions, widestPerfShapeDiameter: number) => {
+const createCanvas = (perfShape: ComplexRopeSegment, options: PerforationOptions): { canvas: HTMLCanvasElement; ctx: CanvasRenderingContext2D } => {
   const canvas = document.createElement('canvas');
-
-  const size = DEFAULT_TEXTURE_SIZE * perforationOptions.scalingFactor;
+  const perfShapeDiameter = perfShape.diameter;
+  const size = perfShapeDiameter * options.scalingFactor;
   canvas.width = size / 2;
   canvas.height = size;
-  const canvasCtx = canvas.getContext('2d');
+  const ctx = canvas.getContext('2d');
 
-  canvasCtx.fillStyle = perforationOptions.yellow;
-  canvasCtx.strokeStyle = perforationOptions.yellow;
-
-  const { packingOpacity } = perforationOptions;
-
-  canvasCtx.fillStyle = perforationOptions.yellow;
-
-  const diameter = widestPerfShapeDiameter - perforationOptions.fracLineLength;
-
-  const xy: [number, number] = [0, canvas.height / 2 - diameter / 2];
-  const wh: [number, number] = [canvas.width, diameter];
-  canvasCtx.save();
-  canvasCtx.globalAlpha = packingOpacity;
-  canvasCtx.fillRect(...xy, ...wh);
-  canvasCtx.restore();
+  return { canvas, ctx };
+};
 
+const createTexture = (canvas: HTMLCanvasElement) => {
```

Comment:
Seems to be related to perforations, let's change the naming to be more related.